platform/windows/joypad_windows.h

```diff
@@ -70,43 +70,30 @@ class JoypadWindows {
 	};
 
 	struct dinput_gamepad {
-		int id;
-		bool attached;
-		bool confirmed;
+		int id = -1;
+		bool attached = false;
+		bool confirmed = false;
 		bool last_buttons[MAX_JOY_BUTTONS];
-		DWORD last_pad;
+		DWORD last_pad = -1;
 
 		LPDIRECTINPUTDEVICE8 di_joy;
 		List<LONG> joy_axis;
 		GUID guid;
 
 		dinput_gamepad() {
-			id = -1;
-			last_pad = -1;
-			attached = false;
-			confirmed = false;
-
 			for (int i = 0; i < MAX_JOY_BUTTONS; i++)
 				last_buttons[i] = false;
 		}
```

Comment:
You can use `bool last_buttons[MAX_JOY_BUTTONS] = {};` and remove the constructor.